workflow-steps/install-node-modules/main.js

```diff
@@ -1,17 +1,50 @@
 const { execSync } = require('child_process');
 const { existsSync, readFileSync, writeFileSync } = require('fs');
 
-const command = getInstallCommand();
+async function main() {
+  const command = getInstallCommand();
 
-if (command) {
-  console.log(`Installing dependencies using ${command.split(' ')[0]}`);
-  console.log(`  Running command: ${command}\n`);
-  execSync(command, { stdio: 'inherit' });
-  patchJest();
-} else {
-  throw new Error(
-    'Could not find lock file. Please ensure you have a lock file before running this command.',
-  );
+  if (command) {
+    console.log(`Installing dependencies using ${command.split(' ')[0]}`);
+    console.log(`  Running command: ${command}\n`);
+
+    const maxRetries = process.env.NX_CLOUD_INPUT_max_retries || 3;
+    let retryCount = 0;
+
+    while (retryCount < maxRetries) {
+      try {
+        execSync(command);
+        patchJest();
+        console.log('Installed dependencies successfully!');
+        break;
+      } catch (e) {
+        retryCount++;
+
+        if (retryCount >= maxRetries) {
```

Comment:
optional, but we could extract the retry mechanism into a utility that you pass a command to `runCommandWithRetry(command: string, maxRetries: number)`